File: src/test/java/se/johannesdahlgren/adventofcode2021/Day10Test.java
package se.johannesdahlgren.adventofcode2021;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

class Day10Test {

  @Test
  void part1Example() {
    Day10 day10 = new Day10("day10Example.txt");
    long syntaxErrorScore = day10.calculateSyntaxErrorScore();
    assertThat(syntaxErrorScore).isEqualTo(26397L);
  }

  @Test
  void part1() {
    Day10 day10 = new Day10("day10.txt");
    long syntaxErrorScore = day10.calculateSyntaxErrorScore();
    assertThat(syntaxErrorScore).isEqualTo(290691L);
  }

  @Test
  void part2Example() {
    Day10 day10 = new Day10("day10Example.txt");
    long syntaxErrorScore = day10.calculateAutoCompleteScore();
    assertThat(syntaxErrorScore).isEqualTo(288957L);
  }

  @Test
  void part2() {
    Day10 day10 = new Day10("day10.txt");
    long syntaxErrorScore = day10.calculateAutoCompleteScore();
    assertThat(syntaxErrorScore).isEqualTo(2768166558L);
  }
}

File: src/test/kotlin/se/johannesdahlgren/aoc22/Day10Test.kt
package se.johannesdahlgren.aoc22

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class Day10Test {
    @Test
    fun part1Example() {
        val day10 = Day10("day10Example")
        assertThat(day10.getSignalStrength()).isEqualTo(13140)
    }

    @Test
    fun part1() {
        val day10 = Day10("day10")
        assertThat(day10.getSignalStrength()).isEqualTo(17380)
    }

    @Test
    fun part2Example() {
        val day10 = Day10("day10Example")
        val expectedResult = """
            ##..##..##..##..##..##..##..##..##..##..
            ###...###...###...###...###...###...###.
            ####....####....####....####....####....
            #####.....#####.....#####.....#####.....
            ######......######......######......####
            #######.......#######.......#######.....
        """.trimIndent()
        assertThat(day10.drawCRTScreen()).isEqualTo(expectedResult)
    }

    @Test
    fun part2() {
        val day10 = Day10("day10")
        val expectedResult = """
            ####..##...##..#..#.####.###..####..##..
            #....#..#.#..#.#..#....#.#..#.#....#..#.
            ###..#....#....#..#...#..#..#.###..#....
            #....#.##.#....#..#..#...###..#....#....
            #....#..#.#..#.#..#.#....#.#..#....#..#.
            #.....###..##...##..####.#..#.####..##..
        """.trimIndent()
        assertThat(day10.drawCRTScreen()).isEqualTo(expectedResult)
    }
}